modules/xmpp/RtxModifier.js

```diff
@@ -0,0 +1,276 @@
+            }
+        });
+        if (fidGroup) {
+          return parseInt(fidGroup.ssrcs.split(" ")[1]);
+        }
+    }
+}
+
+/**
+ * Updates or inserts the appropriate rtx information for primarySsrc with
+ *  the given rtxSsrc.  If no rtx ssrc for primarySsrc currently exists, it will
+ *  add the appropriate ssrc and ssrc group lines.  If primarySsrc already has
+ *  an rtx ssrc, the appropriate ssrc and group lines will be updated
+ * @param {object} videoMLine video mline object that will be updated (in place)
+ * @param {object} primarySsrcInfo the info (ssrc, msid & cname) for the 
+ *  primary ssrc
+ * @param {number} rtxSsrc the rtx ssrc to associate with the primary ssrc
+ */
+function updateAssociatedRtxStream (videoMLine, primarySsrcInfo, rtxSsrc) {
+    logger.info("Updating mline to associate " + rtxSsrc + 
+        " rtx ssrc with primary stream ", primarySsrcInfo.id);
+    let primarySsrc = primarySsrcInfo.id;
+    let primarySsrcMsid = primarySsrcInfo.msid;
+    let primarySsrcCname = primarySsrcInfo.cname;
+
+    let previousAssociatedRtxStream = 
+        getRtxSsrc (videoMLine, primarySsrc);
+    if (previousAssociatedRtxStream === rtxSsrc) {
+        logger.info(rtxSsrc + " was already associated with " +
+            primarySsrc);
+        return;
+    }
+    if (previousAssociatedRtxStream) {
+        logger.info(primarySsrc + " was previously assocaited with rtx " +
+            previousAssociatedRtxStream + ", removing all references to it");
+        // Stream already had an rtx ssrc that is different than the one given,
+        //  remove all trace of the old one
+        videoMLine.ssrcs = videoMLine.ssrcs
+            .filter(ssrcInfo => ssrcInfo.id !== previousAssociatedRtxStream);
+        logger.info("groups before filtering for " + 
+            previousAssociatedRtxStream);
+        logger.info(JSON.stringify(videoMLine.ssrcGroups));
+        videoMLine.ssrcGroups = videoMLine.ssrcGroups
+            .filter(groupInfo => {
+                return groupInfo
+                    .ssrcs
+                    .indexOf(previousAssociatedRtxStream + "") === -1;
+            });
+    }
+    videoMLine.ssrcs.push({
+        id: rtxSsrc,
+        attribute: "cname",
+        value: primarySsrcCname
+    });
+    videoMLine.ssrcs.push({
+        id: rtxSsrc,
+        attribute: "msid",
+        value: primarySsrcMsid
+    });
+    videoMLine.ssrcGroups = videoMLine.ssrcGroups || [];
+    videoMLine.ssrcGroups.push({
+        semantics: "FID",
+        ssrcs: primarySsrc + " " + rtxSsrc
+    });
+}
+/**
+ * End helper functions
+ */
+
+/**
+ * Adds any missing RTX streams for video streams
+ *  and makes sure that they remain consistent
+ */
+export default class RtxModifier {
+    /**
+     * Constructor
+     */
+    constructor () {
+        /**
+         * Map of video ssrc to corresponding RTX
+         *  ssrc
+         */
+        this.correspondingRtxSsrcs = {};
```

Comment:
fixed.  also not crazy about the `set` syntax of `Map`.  to me
`myMap[key] = value;`
is clearer than
`myMap.set(key, value);`
but oh well.